File: GymNotes.Service/Service/UserInfoService.cs
using AutoMapper;
using GymNotes.Entity.Models;
using GymNotes.Models;
using GymNotes.Repository.IRepository;
using GymNotes.Repository.IRepository.User;
using GymNotes.Service.Exceptions;
using GymNotes.Service.IService;
using GymNotes.Service.ViewModels;
using Microsoft.AspNetCore.Identity;
using Microsoft.AspNetCore.Mvc;
using Org.BouncyCastle.Asn1.Ocsp;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace GymNotes.Service.Service
{
  public class UserInfoService : IUserInfoService
  {
    private readonly IMapper _mapper;
    private readonly IUnitOfWork _unitOfWork;

    public UserInfoService(
      IMapper mapper,
      IUnitOfWork unitOfWork)
    {
      _mapper = mapper;
      _unitOfWork = unitOfWork;
    }

    public async Task<ApiResponse> UpdateInstagramURL(StringVm updateURLVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == updateURLVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Instagram = updateURLVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateFacebookURL(StringVm updateURLVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == updateURLVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Facebook = updateURLVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateTwitterURL(StringVm updateURLVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == updateURLVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Twitter = updateURLVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateYoutubeURL(StringVm updateURLVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == updateURLVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Youtube = updateURLVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateDescription(StringVm stringUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == stringUpdateVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Description = stringUpdateVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateDiscipline(StringVm stringUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == stringUpdateVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Discipline = stringUpdateVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateGender(StringVm stringUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == stringUpdateVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Gender = stringUpdateVm.Content;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateYearsOfExperience(NumberVm numberUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == numberUpdateVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.YearsOfExperience = numberUpdateVm.Value;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateHeight(NumberVm numberUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == numberUpdateVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Height = numberUpdateVm.Value;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateBirthday(DateVm dateUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == dateUpdateVm.UserId).FirstOrDefault();

      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.Birthday = dateUpdateVm.Date;

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }

    public async Task<ApiResponse> UpdateIsCoach(StringVm stringUpdateVm)
    {
      var user = _unitOfWork.userRepository.FindByCondition(x => x.Id == stringUpdateVm.UserId).FirstOrDefault();
      if (user == null)
        throw new MyNotFoundException(ApiResponseDescription.USER_NOT_FOUND);

      user.IsCoach = Convert.ToBoolean(stringUpdateVm.Content);

      _unitOfWork.userRepository.Update(user);
      await _unitOfWork.CompleteAsync();

      return new ApiResponse(true);
    }
  }
}

File: GymNotes.Angular/ClientApp/src/app/Core/Services/Http/User/UserInfo.service.ts
import { APIService } from './../API/API.service';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class UserInfoService {

  private CONTROLLER = 'UserInfo/';
  private UPDATE_INSTAGRAM_URL = 'updateInstagramUrl';
  private UPDATE_FACEBOOK_URL = 'updateFacebookUrl';
  private UPDATE_TWITTER_URL = 'updateTwitterUrl';
  private UPDATE_YOUTUBE_URL = 'updateYoutubeUrl';
  private UPDATE_DESCRIPTION = 'updateDescription';
  private UPDATE_DISCIPLINE = 'updateDiscipline';
  private UPDATE_GENDER = 'updateGender';
  private UPDATE_YEARS_OF_EXPERIENCE = 'updateYearsOfExperience';
  private UPDATE_HEIGHT = 'updateHeight';
  private UPDATE_IS_COACH = 'updateIsCoach';
  private UPDATE_BIRTHDAY = 'updateBirthday';

constructor(private API: APIService) { }

  public UpdateInstagramUrl(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_INSTAGRAM_URL, null));
  }

  public UpdateFacebookUrl(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_FACEBOOK_URL, null));
  }

  public UpdateTwitterUrl(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_TWITTER_URL, null));
  }

  public UpdateYoutubeUrl(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_YOUTUBE_URL, null));
  }

  public UpdateDescription(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_DESCRIPTION, null));
  }

  public UpdateDiscipline(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_DISCIPLINE, null));
  }

  public UpdateGender(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_GENDER, null));
  }

  public UpdateTrainingSince(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_YEARS_OF_EXPERIENCE, null));
  }

  public UpdateHeight(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_HEIGHT, null));
  }

  public UpdateBirthday(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_BIRTHDAY, null));
  }

  public UpdateIsCoach(model) {
		return this.API.Post(model, this.API.BuildAddress(this.CONTROLLER, this.UPDATE_IS_COACH, null));
  }
}
